packages/cloud_firestore/cloud_firestore_web/lib/field_value_factory_web.dart

```diff
@@ -0,0 +1,39 @@
+part of cloud_firestore_web;
+
+class FieldValueFactoryWeb implements FieldValueFactory {
+  @override
+  FieldValueInterface arrayRemove(List elements) {
+    final delegate = web.FieldValue.arrayRemove(elements);
+    return FieldValueWeb._(delegate, FieldValueType.arrayRemove, elements);
+  }
+
+  @override
+  FieldValueInterface arrayUnion(List elements) {
+    final delegate = web.FieldValue.arrayUnion(elements);
+    return FieldValueWeb._(delegate, FieldValueType.arrayUnion, elements);
+  }
+
+  @override
+  FieldValueInterface delete() {
+    final delegate = web.FieldValue.delete();
+    return FieldValueWeb._(delegate, FieldValueType.delete, null);
+  }
+
+  @override
+  FieldValueInterface increment(num value) {
+    assert(num is double || num is int, "value can only be double or int");
```

Comment:
Subtle bug here, the assert should be:

```dart
assert(value is double || value is int, "value can only be double or int");
```

(Check the type on `value` not on `num`)

Reply:
Ops 😅  .. good spot